roles/openshift_cli/tasks/main.yml

```diff
@@ -8,11 +8,12 @@
     docker pull {{ openshift.common.cli_image }}:{{ openshift_image_tag }}
   when: openshift.common.is_containerized | bool
 
-- name: Create /usr/local/bin/openshift cli wrapper
-  template:
-    src: openshift.j2
-    dest: /usr/local/bin/openshift
-    mode: 0755
+- set_fact:
+    container_name: "openshift-cli-{{ 100000 | random }}"
+  when: openshift.common.is_containerized | bool
+
+- name: Copy openshift binary out of CLI image for use on the host
+  shell: docker create --name {{ container_name }} {{ openshift.common.cli_image }}:{{ openshift_image_tag }} && docker cp {{ container_name }}:/usr/bin/openshift /usr/local/bin/openshift && docker rm {{ container_name }}
```

Comment:
Also, do we want to check for the existence of the binary and the version of the binary to determine if we need to update the client? I suspect this is getting quite a bit more complex, but for ongoing config it would seem bad to just consistently overwrite the binary.